.npm/postinstall.js

```diff
@@ -1,10 +1,69 @@
-if (!process.env.CI) {
-  const { spawnSync } = require('child_process');
-  const { getExePath } = require('./get-exe');
+const { spawnSync } = require("child_process")
 
+async function install() {
+  if (process.env.CI) {
+    return
+  }
+  const exePath = await downloadBinary()
   // run install
-  spawnSync(getExePath(), ['install', '-f'], {
+  spawnSync(exePath, ["install", "-f"], {
     cwd: process.env.INIT_CWD || process.cwd,
-    stdio: 'inherit',
-  });
+    stdio: "inherit",
+  })
 }
+
+function getDownloadURL() {
+  // Detect OS
+  // https://nodejs.org/api/process.html#process_process_platform
+  let goOS = process.platform
+  let extension = ""
+  if (["win32", "cygwin"].includes(process.platform)) {
+    goOS = "windows"
+    extension = ".exe"
+  }
+
+  // Convert the goOS to the os name in the download URL
+  let downloadOS = goOS === "darwin" ? "macOS" : goOS
+  downloadOS = `${downloadOS.charAt(0).toUpperCase()}${downloadOS.slice(1)}`
+
+  // Detect architecture
+  // https://nodejs.org/api/process.html#process_process_arch
+  let arch = process.arch
+  switch (process.arch) {
+    case "x64": {
+      arch = "x86_64"
+      break
+    }
+    case "x32":
+    case "ia32": {
+      arch = "i386"
+      break
+    }
+  }
+  const version = require("./package.json").version
+
+  return `https://github.com/evilmartians/lefthook/releases/download/v${version}/lefthook_${version}_${downloadOS}_${arch}${extension}`
+}
+
+const { DownloaderHelper } = require("node-downloader-helper")
+const path = require("path")
+
+async function downloadBinary() {
+  // TODO zip the binaries to reduce the download size
+  const downloadURL = getDownloadURL()
+  const extension = path.extname(downloadURL).slice(1)
+  const fileName = `lefthook.${extension}`
+  const binDir = path.join(__dirname, "bin")
+  const dl = new DownloaderHelper(downloadURL, binDir, {
+    fileName,
+    retry: { maxRetries: 5, delay: 50 },
+  })
+  dl.on("end", () => console.log("lefthook binary was downloaded"))
+  await dl.start()
+  return path.join(binDir, fileName)
```

Comment:
On Linux and MacOS downloaded file doesn't have access right to be executed. To fix this, you need to set executable bit here. You can use code like this:

```js
const { chmod } = require("fs/promises");
await chmod(pathToDownloadedBinary, "755");
```

Where `755` is the same access rights as for `bin/index.js` (`755` is the same as `rwxrwxr-x` — owner and group can read, write, and execute, but others can only read and execute)

Reply:
Fixed in https://github.com/evilmartians/lefthook/pull/188/commits/4c46696928589ee3e06edd40031ed012a30f9206